File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode RemoveNthFromEnd(ListNode head, int n) {
        ListNode dummy = new ListNode(), node1 = dummy, node2 = dummy;
        int i = 0;
        dummy.next = head;
        while(i<n)
        {
            node1=node1.next;
            ++i;
        }
        while(node1.next!=null)
        {
            node1=node1.next;
            node2=node2.next;
        }
        node2.next=node2.next.next;
        return dummy.next;
        
    }
}

File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        int size = 0;
        ListNode t = head;
        if (t == null) {
            return head;
        }
        while(t != null) {
            size++;
            t = t.next;
        }
        
        int nth = size-n+1;
        t = head;
        ListNode prev = null;
        
        while (nth-- > 1){
            prev = t;
            t = t.next;
        }
        if(prev == null){
            head = head.next;
            return head;
        }
        prev.next = t.next;
        return head;
    }
}